sdk/azidentity/managed_identity_client.go

```diff
@@ -103,6 +108,11 @@ func (c *managedIdentityClient) authenticate(ctx context.Context, clientID strin
 		return c.createAccessToken(resp)
 	}
 
+	if c.msiType == msiTypeIMDS && resp.StatusCode == 400 {
```

Comment:
What happens in the case of user-assigned identity auth where the user-assigned ID was incorrectly specified, or is that different from IMDS?

Reply:
IMDS responds the same when it can't access a user-assigned or system-assigned identity for any reason. "You specified the wrong ID" gets the same treatment as "managed identity is disabled for this VM". We could distinguish those cases here by assuming that if we have a value for `clientID`, the user intended to authenticate a user-assigned identity; then a 400 indicates misconfiguration of the application or the VM and, arguably, we should return ClientAuthenticationError.

I write "arguably" because we don't do that in other SDKs. I suspect that's an oversight, not by design--I'll have to think through possible consequences a little more. Probably this has just never come up. For a deployed application any managed identity error is typically fatal to authentication because the application is using either ManagedIdentityCredential alone or DefaultAzureCredential, whose remaining credentials integrate with dev tools.